### ISSUE TITLE
Make sure that the constant representation of c-variables is unique

### ISSUE THREAD

**Issue by mudbri** [ravel-net/pyotr#15]
We use the mapping of c-variables to constants while figuring out the mapping of variables in one rule to the other rule while merging rules. Thus, this mapping should make sure that it does not conflict with any constant. Currently, we just use the following line for mapping, which gives distinct values to each variable/c-variable:
```
for i, var in enumerate(self._variables+self._c_variables):
            self._mapping[var] = i
```

**Comment by mudbri** [comment]
Anduo's Idea:
Maybe create a scheme of c-variables (e.g. start the c-variable with rule number (e.g. r1-a, r1-b). We can even introduce the predicate (e.g. r1-l-1-a, r1-l-2-b). This will make sure it is unique and will also make debugging faster

**Comment by mudbri** [comment]
Fixed in commit 489998ff88b69dbec0831d53314af51eeac73b7d
